### ISSUE TITLE
[HLS] Issues with adverts (x-discontinuity)

### ISSUE THREAD

**Issue by matthuisman** [peak3d/inputstream.adaptive#1071]
Hopefully we can use this issue to finally get to the bottom of what causes pauses / stopping when playing back HLS with SSAI / adverts using x-discontinuity

"""
channel 7 works but it seems to get stuck on when an ad break starts. Similar to the issue reported for Pluto TV.

Playback paused for a few seconds at the start of every ad break. It seemed the first 2 ads were local and each had to load, then it reverted to a more generic ad stream. For the past week it used to hang with a percentage sign on screen but it would never restart itself. This was happening with 2 other people as well that I know personally which spurred me to log an issue here. It just so happened when I ran the debug log, it wasn't getting stuck for any more than a few seconds.

Info on debug file:
turned on debug, restarted kodi, went to 7flix and waited for 2 ad breaks. At the start of each ad break, it paused for around 5 seconds, It did this for the first 2 ads in each ad break. I presume it was due to those first 2 ads were local to my region which I haven't noticed before. After 2 ad breaks, I stopped stream and exited Kodi.
"""

I've tried with "use Inputstream HLS for Live Streams" turned off in IPTV AU settings > Playback but it still occurs. It's only an issue with Channel 7 channels"

User provided log:
https://github.com/matthuisman/slyguy.addons/files/10048382/kodi.log

Test stream (australia geolocked)
```
#KODIPROP:inputstream=inputstream.adaptive
#KODIPROP:inputstreamaddon=inputstream.adaptive
#KODIPROP:inputstream.adaptive.manifest_type=hls
https://i.mjh.nz/7-mel6.m3u8
```
I never seem to see these issues on Windows - so maybe Android only issue??

**Comment by matthuisman** [comment]
Some good information here
https://github.com/matthuisman/slyguy.addons/issues/305

**Comment by MajinLugia** [comment]
Will anyone developing this plugin acknowledge this? This issue is beyond annoying and been going on for far too long, I can’t watch any streams that have x-discontinuity ads without them freezing the stream or outright closing the stream. I’m using Android TV.

**Comment by glennguy** [comment]
@MajinLugia what's your expectation here? I can give you the following information:
* The issue has been acknowledged.
* There is a very small team of people actively contributing to this project (2 for a long time, more recently 3)
* The original developers are long gone
* Even with combined knowledge, there are still a lot of areas of the code base that we do not understand
* We are paid $0 per hour to work on this
* This is not the only issue

**Comment by glennguy** [comment]
@matthuisman does this mean this playlist took 6.5 seconds to refresh?
```
2022-11-20 12:29:41.137 T:27813   debug <general>: script.module.slyguy - GET IN: https://csm-e-cen7ausw1live102-0e0415ba9d715c253.tls1.yospace.com/csm/live/191546493/4.m3u8?&&externalId=MEL6&ppId=null&vid=5652241367001&pp=csai-web&yo.pp=aGRudHM9ZXhwPTE2Njg5OTI3MTJ-YWNsPS8qfmhtYWM9OWZmMzVmODZmZjlmMTM4MTVlOWYzY2UzMjZjYzE1ODRjZGU1ZmMyYmU2ZjIyODBlYWFiZTlkY2NiZDIzMWQ4Yg&yo.pdt=sync&platformType=web&used=null&deviceId=71aa8979-75b0-4393-ad93-239d368ed79d&custParams=rc%25253D26%252526seriesid%25253DMHOA&yo.hb=2000&yo.rl=1&ppId=null&vid=5652241367001&optinAdTracking=0&appId=7plus&yo.ap=https%3A%2F%2Fysp.cdn.7livecloud.io%2Fprofile-25-30-60%2F&deviceType=web&optinDeviceType=&videoType=live&yo.ac=false&tvid=null&yo.oh=Y3NtLWUtbjdhdXMtZWIudGxzMS55b3NwYWNlLmNvbQ%3D%3D&yo.sl=4&advertId=null&accountId=5650355166001&rc=26&uaId=null&yo.t.pf=8000&ozid=d9f94c11-e71a-46a8-9c72-fbce1d17b4da&pc=1000&yo.br=true&yo.js=C293DEC8B26EB9F2D9D5C460DCE2E0E0.csm-e-cen7ausw1live102-0e0415ba9d715c253.tls1.yospace.com
2022-11-20 12:29:47.713 T:27813   debug <general>: script.module.slyguy - GET OUT: https://csm-e-cen7ausw1live102-0e0415ba9d715c253.tls1.yospace.com/csm/live/191546493/4.m3u8?&&externalId=MEL6&ppId=null&vid=5652241367001&pp=csai-web&yo.pp=aGRudHM9ZXhwPTE2Njg5OTI3MTJ-YWNsPS8qfmhtYWM9OWZmMzVmODZmZjlmMTM4MTVlOWYzY2UzMjZjYzE1ODRjZGU1ZmMyYmU2ZjIyODBlYWFiZTlkY2NiZDIzMWQ4Yg&yo.pdt=sync&platformType=web&used=null&deviceId=71aa8979-75b0-4393-ad93-239d368ed79d&custParams=rc%25253D26%252526seriesid%25253DMHOA&yo.hb=2000&yo.rl=1&ppId=null&vid=5652241367001&optinAdTracking=0&appId=7plus&yo.ap=https%3A%2F%2Fysp.cdn.7livecloud.io%2Fprofile-25-30-60%2F&deviceType=web&optinDeviceType=&videoType=live&yo.ac=false&tvid=null&yo.oh=Y3NtLWUtbjdhdXMtZWIudGxzMS55b3NwYWNlLmNvbQ%3D%3D&yo.sl=4&advertId=null&accountId=5650355166001&rc=26&uaId=null&yo.t.pf=8000&ozid=d9f94c11-e71a-46a8-9c72-fbce1d17b4da&pc=1000&yo.br=true&yo.js=C293DEC8B26EB9F2D9D5C460DCE2E0E0.csm-e-cen7ausw1live102-0e0415ba9d715c253.tls1.yospace.com (200)
```

Or am I reading IN/OUT from the wrong point of view?

**Comment by matthuisman** [comment]
GET IN called when the proxy receives the request
GET OUT called after the proxy receives a response

So I suspect their server here took 6.5 seconds to respond?
Or the proxy took ages to do its thing?
I really should log directly before the request out so we can tell

**Comment by CastagnaIT** [comment]
PR #1431 should fix this issue, about freeze with ads
if you can please give a feedback

**Comment by CastagnaIT** [comment]
closed as the symptoms seem to be the ones i tested for in the fix PR #1431
if not so, please open a new Issue by using last development omega build
